fs/path.go

```diff
@@ -74,11 +74,14 @@ func sameFile(f1, f2 *currentPath) (bool, error) {
 		// If the timestamp may have been truncated in one of the
 		// files, check content of file to determine difference
 		if t1.Nanosecond() == 0 || t2.Nanosecond() == 0 {
-			if f1.f.Size() > 0 {
-				eq, err := compareFileContent(f1.fullPath, f2.fullPath)
-				if err != nil || !eq {
-					return eq, err
-				}
+			var eq bool
+			if (f1.f.Mode() & os.ModeSymlink) == os.ModeSymlink {
+				eq, err = compareSymlinkTarget(f1.fullPath, f2.fullPath)
```

Comment:
Is this not an issue if f1 is not a symlink but f2 is a symlink?

Reply:
Why the nanosecond check at line 76 all? Shouldn't it be `&&` if both layers were extracted from tar(that use case doesn't exist in builder for example)? If only one of them has nanoseconds then it is already always a change.

Reply:
@dnephin `compareSysStat` should catch that case already

Reply:
@tonistiigi I believe we have had this discussion before and came to the conclusion that `&&` is more correct.